src/analysis/ppx_expand.ml

```diff
@@ -0,0 +1,155 @@
+type ppx_kind =
+  | Expr of Parsetree.expression
+  | Sig_item of Parsetree.signature_item
+  | Str_item of Parsetree.structure_item
```

Comment:
In the LSP implementation they handle only two cases `Ppx_expr` and `Ppx_typedef_attr`. Could you explain to me the difference ?

Reply:
So in the LSP implementation, `Ppx_expr` is to handle PPX extension nodes `[%foo]` and `Ppx_typedef_attr` is for PPX derivers on `type_declarations`. So this was a bit shallow to me as it will only work for `Type declarations`, but we need the expand feature to work on all PPX nodes. and this includes:
- type declaration in structure and signature
- type extension in structure and signature
- exception in structure and signature
- module type declaration in structure and signature

So this PR handles all the different nodes, and not just `type_declarations`

Reply:
I'll update the `ocaml-lsp` PR to handle all these nodes as well because currently it will only trigger `ppx-expand` on Merlin if it's a `type_declaration` or `expression`.